### ISSUE TITLE
Use proper logging instead of prints

### ISSUE THREAD

**Issue by pmayd** [pmayd/open-discourse#5]
We should implement a proper logging mechanism and extend the basic prints with meaningful log messages as well as persistent log files

**Comment by erwee6** [comment]
**Preliminaries**

Certain basic agreements on logging should be made, e.g.:

- standardized format for logging messages
- where should the logs be written to
- guideline for the definition of logging.level
- other ?

**Comment by tifa365** [comment]
@erwee6 You're welcome to chime in but my script only had one logging statement, with the syntax I just follow the print() statement that I refactored into a log.  

Which loggin level to use is a good question. I went with the standard. `logging.basicConfig(level=logging.INFO)`

@pmayd What do you think?

**Comment by erwee6** [comment]
@tifa365: in which folder your logging file is located and should every script use the same logfile. I would prefer this.

**Comment by pmayd** [comment]
Hi,
Logging should not be alone to the console but into files so it is not enough to just use logging, you need to setup a proper logging with two handles, one for console and one for files. And no the files should not log all equally, for each script I think it's best to have one log file. But yes then each logger for each file can have the same base config . 1

**Comment by pmayd** [comment]
Hi,
Logging should not be alone to the console but into files so it is not enough to just use logging, you need to setup a proper logging with two handles, one for console and one for files. And no the files should not log all equally, for each script I think it's best to have one log file. But yes then each logger for each file can have the same base config . 

@tifa365 regarding log level it doesn't really matter but as this is not a library but a pipeline that is still heavily developed the log level should be DEBUG to log everything. Normally with logging.debug you would log simple status calls like script started, number if files found etc. of course all of that could also be info but I like to have the possibility do reduce the log noise by switching to log level INFO

**Comment by erwee6** [comment]
@pmayd I don‘t understand this fully. When all scipts are running in the process chain, I have to look into n logfiles?

**Comment by pmayd** [comment]
Yes exactly. You could configure the logger to write to the same file again but why should we want one mega log file? Normally, each process and each script is isolated from the others and doing its part so when I want to know when or where something happens I would like to check that log file for that step. At least that is what I would expect and how I am using log files, each step and process has its own log file. Later we can combine these if we are interested in a total log. But if you want to try the other way around you can also try to write into a single log file but then you have to work with different log names or a lot of debug messages to make the start and end of each script clear. I am not totally against it but I would go for the simpler solution of having exactly one log file per main script

**Comment by erwee6** [comment]
@pmayd: @tifa365 

Where should the individual logs should be stored? 

Perplexity suggests: 
1. Dedicated Logs Directory at Project Root
2. Logs Subdirectory in src Folder
3. Configurable Solution (but then a directory must be named)

**Comment by pmayd** [comment]
I don't think we need a config here, it is not for the user or any other person than us. I usually go with a log folder in root, but here I would keep it close to the data folder. Originally, data was on the root level but I think I moved it under python and that is also where to put the log folder in my opinion. The src folder is definitely the wrong place in my opinion as it is specifically about source code files so you would never look for logs in there so I am against this but next to data under python folder should be ideal

**Comment by erwee6** [comment]
We are talking past each other. The setup of the logger and the handler must take place somewhere. I do this in a code snippet logging_config.py. I can then import it if necessary.

**Comment by pmayd** [comment]
I know that and we do not speak part each other, as I was referencing to your last option that you mentioned, configuring the log location in a config. We don't need that, we are not writing a library for others to be used so we can simply define one location. That is why I said we don't need a config to configure the location of our logger :)

**Comment by erwee6** [comment]
Obviously I don't understand your point or vice versa. Let's discuss this in the meeting.
